### PR TITLE
U4X-658: Make sure concepts created by the attachment module are persisted to those created by UgandaEMR.

### PR DESCRIPTION
**Pull request by slubwama**
https://metsprogramme.atlassian.net/browse/U4X-658

follow on for PR https://github.com/METS-Programme/openmrs-module-ugandaemr/pull/542

### CODE REVIEW

File: api/src/main/resources/liquibase.xml [hunk 1]
```diff
@@ -1020,6 +1020,32 @@ VALUES
             WHERE pp.program_id IN (5,6,7,8,9);
         </sql>
     </changeSet>
+
+	<changeSet id="ugandaemr-20240610-1655" author="slubwama">
+		<preConditions onFail="CONTINUE">
+			<sqlCheck expectedResult="1">
+				select count(*) from concept where uuid=="7cac8397-53cd-4f00-a6fe-028e8d743f8e";
```
Comment (akileng56) [inline api/src/main/resources/liquibase.xml]: Wrong MySQL syntax `select count(*) from concept where uuid=="7cac8397-53cd-4f00-a6fe-028e8d743f8e";`

Should be `select count(*) from concept where uuid="7cac8397-53cd-4f00-a6fe-028e8d743f8e";`

Single equal...

File: api/src/main/resources/liquibase.xml [hunk 2]
```diff
@@ -1020,6 +1020,32 @@ VALUES
...
+		<sql>
+			update concept_name set concept_name.uuid="552e757e-b072-4e68-a10d-b7db7efcd5ad", concept_name.concept_name_id=198844, concept_name.concept_id=186351  where concept_id in (select concept.concept_id from  concept where concept.uuid="7cac8397-53cd-4f00-a6fe-028e8d743f8e");
+			update concept set concept_id=186351 where uuid="7cac8397-53cd-4f00-a6fe-028e8d743f8e";
+		</sql>
+	</changeSet>
+
+	<changeSet id="ugandaemr-20240610-1656" author="slubwama">
+		<preConditions onFail="CONTINUE">
+			<sqlCheck expectedResult="1">
+				select count(*) from concept where uuid=="42ed45fd-f3f6-44b6-bfc2-8bde1bb41e00";
```
Comment (akileng56) [inline api/src/main/resources/liquibase.xml]: Same here

**Review by akileng56 — CHANGES_REQUESTED** [review]

File: api/src/main/resources/liquibase.xml [hunk 3]
```diff
@@ -1020,6 +1020,32 @@ VALUES
...
+
+	<changeSet id="ugandaemr-20240610-1655" author="slubwama">
+		<preConditions onFail="CONTINUE">
+			<sqlCheck expectedResult="1">
+				select count(*) from concept where uuid="7cac8397-53cd-4f00-a6fe-028e8d743f8e";
+			</sqlCheck>
+		</preConditions>
+		<comment>Change Image attachment to known concept metadata in UgandaEMR</comment>
+		<sql>
+			update concept_name set concept_name.uuid="552e757e-b072-4e68-a10d-b7db7efcd5ad", concept_name.concept_name_id=198844, concept_name.concept_id=186351  where concept_id in (select concept.concept_id from  concept where concept.uuid="7cac8397-53cd-4f00-a6fe-028e8d743f8e");
```
Comment (akileng56) [inline api/src/main/resources/liquibase.xml:1032]: `Caused by: liquibase.exception.DatabaseException: Cannot add or update a child row: a foreign key constraint fails (`dev-plus`.`concept_name`, CONSTRAINT `concept_name_ibfk_1` FOREIGN KEY (`concept_id`) REFERENCES `concept` (`concept_id`)) [Failed SQL: (1452) update concept_name set concept_name.uuid="552e757e-b072-4e68-a10d-b7db7efcd5ad", concept_name.concept_name_id=198844, concept_name.concept_id=186351  where concept_id in (select concept.concept_id from  concept where concept.uuid="7cac8397-53cd-4f00-a6fe-028e8d743f8e")]
`

**Review by akileng56 — CHANGES_REQUESTED** [review]

File: api/src/main/resources/liquibase.xml (hunk 3 above)
Comment (slubwama) [reply api/src/main/resources/liquibase.xml:1032]: @akileng56 reset concept dic then restart

**Review by akileng56 — APPROVED** [review]

**Pull request merged**